usecases/replica/coordinator.go

```diff
@@ -40,28 +47,37 @@ type (
 		Class    string
 		Shard    string
 		TxID     string // transaction ID
+		// wait twice this duration for the first Pull backoff for each host
+		pullBackOffPreInitialInterval time.Duration
+		pullBackOffMaxElapsedTime     time.Duration // stop retrying after this long
 	}
 )
 
 // newCoordinator used by the replicator
 func newCoordinator[T any](r *Replicator, shard, requestID string, l logrus.FieldLogger,
 ) *coordinator[T] {
 	return &coordinator[T]{
-		Client:   r.client,
-		Resolver: r.resolver,
-		log:      l,
-		Class:    r.class,
-		Shard:    shard,
-		TxID:     requestID,
+		Client:                        r.client,
+		Resolver:                      r.resolver,
+		log:                           l,
+		Class:                         r.class,
+		Shard:                         shard,
+		TxID:                          requestID,
+		pullBackOffPreInitialInterval: defaultPullBackOffInitialInterval / 2,
```

Comment:
**suggestion:** is `defaultPullBackOffInitialInterval / 2` gonna be always the case ? because if so we can just define `defaultPullBackOffInitialInterval` with this value directly. 

if it's configurable maybe we pass it as param

Reply:
Also going to look at this more in a follow up PR if that's ok, it's a bit convoluted in the current code but my thought process was that we call NextBackoff before we actually use the backoff, so whatever the default is/user specifies should be cut in half when saved, since it will have NextBackoff called on it before performing the first backoff.